view/frontend/web/js/view/payment/adyen-methods.js

```diff
@@ -76,13 +76,20 @@ define(
                     shippingAddressCountry = quote.shippingAddress().countryId;
                     fullScreenLoader.startLoader();
                     // Retrieve adyen payment methods
-                    adyenPaymentService.retrievePaymentMethods().done(function(paymentMethods) {
-                        paymentMethods = JSON.parse(paymentMethods);
-                        adyenPaymentService.setPaymentMethods(paymentMethods);
+                    try {
+                        adyenPaymentService.retrievePaymentMethods().done(function (paymentMethods) {
+                            paymentMethods = typeof paymentMethods === 'string' ? JSON.parse(paymentMethods) : paymentMethods;
```

Comment:
It happened on a 2.3.1. Not quite sure how to reply to this. I did get a JSON as an argument, which is why i had to avoid parsing it, if it was an array.